src/java.base/share/classes/jdk/internal/util/Architecture.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2018, 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.internal.util;
+
+import jdk.internal.vm.annotation.ForceInline;
+
+/**
+ * System architecture enum values.
+ * Each architecture has a matching {@code public static boolean isXXX()} method
+ * that is true when running on that architecture.
+ * The enum values correspond to architectures as named in the build system.
+ * See the values of `OPENJDK_TARGET_ARCH_OSARCH`.
+ * The order and ordinal values must match the values defined in OperatingSystemProps.java.template.
+ */
+public enum Architecture {
+    X64(),        // Represents AMD64 and X86_64
+    X86(),
+    IA64(),
+    ARM(),
```

Comment:
```suggestion
    ARM32(),
```

Reply:
I'm inclined to drop the enum value and isArm method because they are legacy and are not being used in OpenJDK sources.  This could also apply to IA64. 
They can be added if needed and otherwise are just unused bulk.